InsightLogParser.Client/Spider.cs

```diff
@@ -573,7 +581,11 @@ private void RouteHandleSolved(int puzzleId)
             _puzzleRouter.AddSolved(puzzleId);
             var current = _puzzleRouter.CurrentNode();
             if (current == null) return;
-            if (current.Value.Node.Puzzle.KrakenId != puzzleId) return;
+            if (current.Value.Node.Puzzle.KrakenId != puzzleId)
```

Comment:
This code handles the case where you're in a route, but want to find the match to a Matchbox that isn't necessarily in your route.  After you solve the Matchbox, you want to be back on the route, and this code handles that.

Reply:
Just as an fyi. I missed it in my code review, but this actually introduces a bug where solving any puzzle will advance the route to the next node.

I've fixed it on my side so no need to worry about it, just wanted to let you know

Reply:
Ah, _that's_ where I did that...  I didn't realize I had bopped the return.  Sorry about that. :)  I did find it and put it back, but haven't had a chance to PR it back in.  Good find.